src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -186,11 +197,13 @@ def runner(
                     self.blue_full_system_proto_unix_io.send_proto(
                         RobotStatus, robot_status
                     )
-                    # We need this blocking get call to synchronize the running speed of world and primitives
-                    # Otherwise, we end up with behaviour that doesn't simulate what would happen in the real world
-                    self.primitive_set_buffer.get(
-                        block=True, timeout=WORLD_BUFFER_TIMEOUT, return_cached=False
-                    )
+
+            # get the time difference after we get the primitive (after any blocking that happened)
+            primitive_block_time = primitive_block_time - time.time()
```

Comment:
This logic is backwards
```suggestion
            primitive_block_time = time.time() - primitive_block_time
```